rfd/0144-client-tools-updates.md

```diff
@@ -376,32 +383,45 @@ a downgrade.
 6. User logs back into Cluster B (after certs expire) → the app prompts for 
 an upgrade again.
 
-This can create a feedback loop of conflicting version updates and repeated 
-restarts. One potential improvement is to stop automatic downloads when multiple
-client versions are detected, giving more control to the user.
-However, this would effectively make updates opt-in, which could delay important 
-fixes or features.
+This can create a feedback loop of constant updates and repeated restarts.
 
 Proposed Solution:
-* The app will read the client tool versions from all connected clusters, 
-and install the latest one by default.
-* If multiple conflicting versions are detected, the auto-update widget will 
-display a warning when logging into a cluster that does not match the currently 
-installed version: "App version is managed by another cluster." Clicking "More" 
-will take the user to the detailed view.
-* In detailed view, the user will be able to choose which cluster should 
-manage updates (a mechanism similar to an update channel), the UI will look as 
-follows:
-> [ ] Use the latest version from your clusters
+* The app will read the client tool versions and minimum client version from all 
+connected clusters, and will try to find the most compatible version (fulfilling 
+our compatibility promise).
+* If no compatible version is found, the auto-update mechanism will stop working
+until the user selects a cluster managing updates in the detailed view.
+The selected cluster will be stored in the app state and cleared when the user 
+logs out from that cluster.
+The UI will look as follows:
+> App updates are disabled
+> 
+> Your clusters require incompatible client versions.
+> To enable app updates, select the cluster managing them manually.
```

Comment:
```suggestion
> To enable app updates, select which cluster should manage them.
```

The current version is a bit vague language wise (is the cluster managing the updates manually or is the user asked to manually select the cluster?).